geo/src/algorithm/intersects/collections.rs

```diff
@@ -13,6 +14,7 @@ where
     Rect<T>: Intersects<G>,
     Polygon<T>: Intersects<G>,
     MultiPolygon<T>: Intersects<G>,
+    G: BoundingRect<T>,
```

Comment:
Note that I had to add some new bounds to some of our intersection traits. 

This shouldn't affect any of our implementations because BoundingBox was already implemented for all geometry types.